From e9dc8e92389aa997d0b9b924289c079e6ad7cf54 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Wed, 26 Apr 2017 15:22:00 -0700
Subject: [PATCH] DOCKER-1043 cli-links.test.js failure due to nginx:latest no
 longer exposing port 443 Reviewed by: Todd Whiteman
 <todd.whiteman@joyent.com> Approved by: Todd Whiteman
 <todd.whiteman@joyent.com>

---
 test/images/test-nginx/Dockerfile                |  3 +++
 .../{test-run-ports => test-nginx}/Makefile      |  2 +-
 test/images/test-nginx/README.md                 |  7 +++++++
 test/images/test-run-ports/Dockerfile            | 16 ----------------
 test/images/test-run-ports/README.md             |  2 --
 test/integration/cli-links.test.js               | 11 +++--------
 test/integration/cli-run-ports.test.js           |  2 +-
 7 files changed, 15 insertions(+), 28 deletions(-)
 create mode 100644 test/images/test-nginx/Dockerfile
 rename test/images/{test-run-ports => test-nginx}/Makefile (91%)
 create mode 100644 test/images/test-nginx/README.md
 delete mode 100644 test/images/test-run-ports/Dockerfile
 delete mode 100644 test/images/test-run-ports/README.md

diff --git a/test/images/test-nginx/Dockerfile b/test/images/test-nginx/Dockerfile
new file mode 100644
index 0000000..d941703
--- /dev/null
+++ b/test/images/test-nginx/Dockerfile
@@ -0,0 +1,3 @@
+FROM nginx:1.13.0-alpine
+EXPOSE 80 443
+CMD ["nginx", "-g", "daemon off;"]
diff --git a/test/images/test-run-ports/Makefile b/test/images/test-nginx/Makefile
similarity index 91%
rename from test/images/test-run-ports/Makefile
rename to test/images/test-nginx/Makefile
index 762c2b0..b48e320 100644
--- a/test/images/test-run-ports/Makefile
+++ b/test/images/test-nginx/Makefile
@@ -8,7 +8,7 @@
 # Copyright 2017 Joyent, Inc.
 #
 
-NAME=joyentunsupported/test-run-ports
+NAME=joyentunsupported/test-nginx
 VERSION=1.0.0
 
 all:
diff --git a/test/images/test-nginx/README.md b/test/images/test-nginx/README.md
new file mode 100644
index 0000000..8a476d0
--- /dev/null
+++ b/test/images/test-nginx/README.md
@@ -0,0 +1,7 @@
+A simple and *stable* nginx image for use in sdc-docker.git
+tests. Before this we often used 'nginx:latest', and hit at
+least one bug there it changed its exposed ports that broke
+our tests.
+
+- exposes TCP ports 80 and 443
+- uses the nginx-alpine image to be smaller
diff --git a/test/images/test-run-ports/Dockerfile b/test/images/test-run-ports/Dockerfile
deleted file mode 100644
index d6c2c27..0000000
--- a/test/images/test-run-ports/Dockerfile
+++ /dev/null
@@ -1,16 +0,0 @@
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright 2017 Joyent, Inc.
-#
-
-# We use "nginx" so we can ports actually in use. We use the "-alpine" version
-# because it is much smaller.
-FROM nginx:1.13.0-alpine
-LABEL description="a test image for https://github.com/joyent/sdc-docker/blob/master/test/integration/cli-run-ports.test.js"
-EXPOSE 80 443
-CMD ["nginx", "-g", "daemon off;"]
diff --git a/test/images/test-run-ports/README.md b/test/images/test-run-ports/README.md
deleted file mode 100644
index 1f3970c..0000000
--- a/test/images/test-run-ports/README.md
+++ /dev/null
@@ -1,2 +0,0 @@
-A simple image that exposes TCP ports 80 and 443 for use by
-test/integration/cli-run-ports.test.js.
diff --git a/test/integration/cli-links.test.js b/test/integration/cli-links.test.js
index 5fd13d6..00ed7ca 100644
--- a/test/integration/cli-links.test.js
+++ b/test/integration/cli-links.test.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -18,20 +18,15 @@ var test = require('tape');
 var vasync = require('vasync');
 
 
-
 // --- Globals
 
-
 var CLIENTS = {};
 var CONTAINER_PREFIX = 'sdcdockertest_link_';
-
-
-// --- Helpers
+var TEST_IMAGE = 'joyentunsupported/test-nginx:1.0.0';
 
 
 // --- Tests
 
-
 test('setup', function (tt) {
 
     tt.test('DockerEnv: alice init', cli.init);
@@ -74,7 +69,7 @@ test('linked env', function (tt) {
     var nginxName = CONTAINER_PREFIX + 'nginx';
     tt.test('linked env: create custom nginx -p 80:80', function (t) {
         cli.run(t, { args: '-d --name ' + nginxName + ' -e FOO=BAR -e BAT=BAZZA'
-                    + ' -p 80:80 nginx' });
+                    + ' -p 80:80 ' + TEST_IMAGE });
     });
 
 
diff --git a/test/integration/cli-run-ports.test.js b/test/integration/cli-run-ports.test.js
index a4828ed..4f8ede2 100644
--- a/test/integration/cli-run-ports.test.js
+++ b/test/integration/cli-run-ports.test.js
@@ -26,7 +26,7 @@ var test = require('tape');
 
 // --- Globals
 
-var TEST_IMAGE = 'joyentunsupported/test-run-ports:1.0.0';
+var TEST_IMAGE = 'joyentunsupported/test-nginx:1.0.0';
 var MAX_PORTS_PER_RULE = 8;
 var FWRULE_VERSION = 1;
 
-- 
2.21.0

